File: android/http-core/src/main/java/com/caldremch/http/core/abs/IConvertStrategy.java
package com.caldremch.http.core.abs;

import com.caldremch.http.core.framework.base.IBaseResp;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.io.InputStream;
import java.lang.reflect.Type;

/**
 * Created by Leon on 2022/10/30.
 */
public interface IConvertStrategy {
    <T> boolean isStreamConvert(@NotNull Type cls);
    <T> T convertStream(@Nullable InputStream inputStream);
    <T> IBaseResp<T> convertCommon(@NotNull String str, @NotNull Type cls);
}

File: http-core/src/main/java/com/caldremch/http/core/abs/IConvertStrategy.kt
package com.caldremch.http.core.abs

import java.io.InputStream

/**
 * Created by Leon on 2022/7/5
 */
interface IConvertStrategy{
    fun <T> isStreamConvert(clz: Class<T>):Boolean
    fun <T> convertStream(inputStream: InputStream?):T
    fun <T> convertCommon(bodyString:String, clz: Class<T>):T
}